src/main/src/main/java/org/geoserver/catalog/TransformFeatureTypeCallback.java

```diff
@@ -72,7 +73,8 @@ private Definition toDefinition(AttributeTypeInfo ati) {
             String name = ati.getName();
             Expression source = ECQL.toExpression(ati.getSource());
             Class<?> binding = ati.getBinding();
-            return new Definition(name, source, binding);
+            String description = StringUtils.trimToNull(ati.getDescription());
+            return new Definition(name, source, binding, null, description);
```

Comment:
GeoTools AttributeTypeInfo.getDescription() is already working with an InternationalString here.